spark_log_parser/parsing_models/application_model_v2.py

```diff
@@ -115,16 +135,30 @@ def save_to_s3(self, saveDat, filepath, compress):
 
 SparkApplicationRawDataType = TypeVar("SparkApplicationRawDataType")
 SparkApplicationLoaderKey = TypeVar("SparkApplicationLoaderKey")
+SparkApplicationClass = TypeVar("SparkApplicationClass", bound=SparkApplication)
```

Comment:
Using a `TypeVar` with a bound plays better with Python type-checkers than typing an abstract method as returning a `SparkApplication`, but then actually returning a sub-class in your concrete implementation

Reply:
Indeed!